packages/@lwc/template-compiler/src/codegen/static-element-serializer.ts

```diff
@@ -74,20 +89,21 @@ function serializeAttrs(element: Element): string {
         })
         .forEach(collector);
 
-    return (attrs.length > 0 ? ' ' : '') + attrs.join(' ') + (hasClassAttr ? '${2}' : '${3}');
+    return attrs.join('') + (hasClassAttr ? '${2}' : '${3}');
```

Comment:
I modified the spacing here so that it's easier to join at runtime, the main issue before was removing the leading space when running on a browser (serialization not required).